main.yml

```diff
@@ -6,6 +6,16 @@
 - import_playbook: setup_undercloud.yml
   tags:
     - always
+  when: virtual_uc != true
+
+- import_playbook: virtual_undercloud.yml
+  tags:
+    - always
+  when: virtual_uc == true
+
+- import_playbook: add_undercloud_to_inventory.yml
```

Comment:
If  virtual_uc is set to true, then this playbook is imported as part of setup_undercloud.yml. In that case, add_undercloud_to_inventory.yml is run a second time. Maybe have a when condition to this import?

Reply:
Nice catch. @masco you forgot to remove add_undercloud_to_inventory.yml from setup_undercloud.yml.

Reply:
removed